builder/solver.go

```diff
@@ -143,6 +143,7 @@ func (s *solver) solveDockerTar(ctx context.Context, localDirs map[string]string
 			case <-ctx.Done():
 				// Close read pipe on cancels, otherwise the whole thing hangs.
 				pipeR.Close()
+				return
```

Comment:
BTW, do we even need the `for` loop? The select basically waits for ctx.Done() indefinitely.

Reply:
hah, that's a great point! we probably dont. I'll simplify this.